File: packages/deterministic-mapping/src/mappings/activity.ts
import { DIMENSIONS, DOMAINS, INTENTION_TYPES, CanonicalFact, MappingResult } from '../types';

export function mapActivityFacts(facts: CanonicalFact[]): MappingResult | null {
  const activityFact = facts.find((f) => f.dimension === DIMENSIONS.ACTIVITY_TYPE);
  if (!activityFact) return null;

  return {
    domain: DOMAINS.ACTIVITY,
    type: INTENTION_TYPES.ACTIVITY_LOG,
    attributes: { activityType: activityFact.value },
    confidence: activityFact.confidence,
  };
}

File: packages/mcp-intent-gateway/src/mcp_intent_gateway/mapping/activity.py
"""Activity mapping function."""

from typing import Any

from ..domain.constants import Dimension, Domain, IntentionType
from ..models.canonical_fact import CanonicalFact


def map_activity_facts(facts: list[CanonicalFact]) -> dict[str, Any] | None:
    """
    Map activity-related facts to a MappingResult.

    Args:
        facts: List of canonical facts to process

    Returns:
        Mapping result dict or None if no activity facts found
    """
    activity_fact = next(
        (f for f in facts if f.dimension == Dimension.ACTIVITY_TYPE),
        None,
    )

    if not activity_fact:
        return None

    return {
        "domain": Domain.ACTIVITY,
        "type": IntentionType.ACTIVITY_LOG,
        "attributes": {"activityType": activity_fact.value},
        "confidence": activity_fact.confidence,
    }
